server/controllers/login/login.js

```diff
@@ -0,0 +1,37 @@
+const { compare } = require('bcryptjs');
+const jwt = require('jsonwebtoken');
+const checkUser = require('../../database/queries/checkUser');
+
+exports.post = (request, response, next) => {
+  const { user_name, password } = request.body;
+  let payload;
+  checkUser(user_name)
+    .then((res) => {
+      if (res.rows[0]) {
+        const { user_name, role } = res.rows[0];
+        payload = {
+          user_name,
+          role,
+        };
+        return compare(password, res.rows[0].password);
+      }
+    })
+    .then((isHashed) => {
+      if (!isHashed) {
+        next({ code: 401 });
+      } else {
+        const token = jwt.sign(payload, process.env.SECRET);
+        response.cookie('jwt', token, { maxAge: 60 * 60 * 24 }, { httpOnly: true });
+        response.status(200).send({
+          error: null,
+          data: {
+            code: 200,
+            msg: 'Success Login',
```

Comment:
@Angham116 If you want, you could return the user info (user_name, role, maybe even id) in this response too.  It's often convenient for the client code to get data about the current user when they successfully login.

Reply:
@Angham116 I just saw #185 , you already did this haha, sorry 🤦‍♂️